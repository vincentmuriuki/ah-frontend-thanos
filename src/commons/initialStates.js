const initialState = {
  articleReducer: {
    article: {
      id: 0,
      tag_list: [],
      slug: '',
      title: '',
      description: '',
      body: '',
      created_at: '',
      updated_at: '',
      image_url: '',
      audio_url: '',
      read_time: '',
      likes: 0,
      dislikes: 0,
      likeDislikeStatus: '',
      views_count: '',
      rating: 0,
      author: {
        id: 0,
        username: '',
        email: '',
      },
    },
    articles: [],
    errorMessage: '',
  },
  deleteArticleReducer: {},
  socialLoginReducer: { isLoggedIn: false },
  loginReducer: { errorMessage: '', successMessage: '', user_details: '' },
  userReducer: {
    freshUser: { email: '', password: '', username: '' },
  },
};

export const sampleArticle = {
  id: 1,
  tag_list: ['tag'],
  slug: 'title-sample',
  title: 'title sample',
  description: 'desc',
  body: 'body',
  created_at: '12-12-2018',
  updated_at: '12-12-2018',
  image_url: 'http://test.png',
  audio_url: 'http://test.mp3',
  read_time: '0 min',
  likes: 12,
  dislikes: 2,
  likeDislikeStatus: 'dislike',
  views_count: 0,
  rating: 0,
  author: {
    id: 2,
    username: 'janedoe',
    email: 'janedoe@gmail.com',
  },
};

export const sampleArticle2 = {
  ...sampleArticle,
  likeDislikeStatus: 'like',
};

export const initialStateWithSample = {
  article: sampleArticle,
};

export const initialStateWithSample2 = {
  article: sampleArticle2,
};

export default initialState;
