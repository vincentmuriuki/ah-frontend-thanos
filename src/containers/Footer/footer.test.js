import React from 'react';
import { shallow, mount } from 'enzyme';
import { Footer, CreateImage } from '.';
import { sampleListOfArticles } from '../../commons/initialStates';

describe('Footer tests', () => {
  let wrapper;
  let toTop;

  beforeEach(() => {
    toTop = jest.fn();
    wrapper = mount(<Footer articles={sampleListOfArticles} toTop={toTop} />);
  });

  test('testing footer', () => {
    const footer = shallow(
      <Footer articles={sampleListOfArticles} />,
    );
    expect(footer).toMatchSnapshot();
  });

  test('testing create image', () => {
    const createImageFooter = shallow(
      <CreateImage />,
    );
    expect(createImageFooter).toMatchSnapshot();
  });

  test('testing div', () => {
    expect(wrapper.find('.footer').hasClass('row'));
  });

  test('click go to top button', () => {
    const fakeEvent = jest.fn();
    const spy = jest.spyOn(wrapper.instance(), 'toTop');
    wrapper.instance().toTop(fakeEvent);
    expect(spy).toHaveBeenCalled();
  });
});
