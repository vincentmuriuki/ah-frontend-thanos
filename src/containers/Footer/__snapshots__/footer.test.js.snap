// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests testing create image 1`] = `
<div
  className="image-card"
>
  <a
    href="/article/undefined"
  >
    <img
      alt="article image1"
    />
  </a>
</div>
`;

exports[`Footer tests testing footer 1`] = `
<div
  className="footer"
>
  <div
    className="carousel"
  >
    <div
      className="carousel slide"
      data-ride="carousel"
      id="carouselExampleIndicatorsFooter"
    >
      <div
        className="carousel-inner"
      >
        <div
          className="carousel-item active"
        >
          <div
            className="row"
          >
            <div
              className="image-card"
              key="1"
            >
              <a
                href="/article/1"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="2"
            >
              <a
                href="/article/2"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="3"
            >
              <a
                href="/article/3"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="4"
            >
              <a
                href="/article/4"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="5"
            >
              <a
                href="/article/5"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="6"
            >
              <a
                href="/article/6"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="7"
            >
              <a
                href="/article/7"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
            <div
              className="image-card"
              key="8"
            >
              <a
                href="/article/8"
                title="title sample"
              >
                <img
                  alt="article image1"
                  src="http://test.png"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer-bottom"
  >
    <div
      className="copyright"
    >
      <span>
        Â© 2018 Authors Haven
      </span>
    </div>
    <div
      className="go-to-top"
    >
      <span>
        <button
          className="btTop"
          id="btTop"
          onClick={[Function]}
          type="button"
        >
          Top
        </button>
      </span>
    </div>
    <div
      className="nav-bottom"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            href="/articles"
          >
            ARTICLE
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            CONTACT
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
